core/src/main/scala/package.scala

```diff
@@ -40,4 +40,15 @@ package object sfs extends sfs.api.Api {
       case Failure(t) => l(t)
     }
   }
+
+  trait Functor[F[_]] {
+    def map[A, B](f: A => B): F[A] => F[B]
+  }
+  object Functor {
+    implicit def forMappable[A, M[A] <: { def map[B](f: A => B): M[B] }]: Functor[M] =
```

Comment:
Here I'm using structural types to provide a type class. The instance states: I can provide an instance of `Functor` for every `M[A]` that has a `def map[B](f: A => B): M[B]`

What is the problem you see?